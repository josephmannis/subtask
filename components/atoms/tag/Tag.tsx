import styled from 'styled-components/native';
import React from 'react';
import {TouchableHighlight} from 'react-native'

const Base = styled.View`
    background-color: #3BA9ED;
    border-radius: 22px;
    display: flex;
    align-items: center;
    padding: 12px 20px;
    align-self: flex-start;
    flex-wrap: nowrap;
    min-width: 100px;

    shadow-color: black;
    shadow-radius: 12px;
    shadow-opacity: .15;
    elevation: 12;
`

const Label = styled.Text`
    color: white;
    font-weight: bold;
    font-size: 16px;
`

const Touchable = styled.TouchableHighlight`
    border-radius: 22px;
`

interface TagProps {
    text: string;
    onPress?: () => void;
}

const Tag: React.FC<TagProps> = props => {
    return (
        <Touchable onPress={props.onPress}>
                <Base>
                    <Label>
                        {props.text}
                    </Label>
                </Base>
        </Touchable>
    )
}

export default Tag;