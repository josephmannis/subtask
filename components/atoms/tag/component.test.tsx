import React from 'react';

import renderer from 'react-test-renderer';
import Tag from './Tag';
test('Renders', () => {
    const tree = renderer.create(<Tag text={'Test tags!'}/>).toJSON();
    expect(tree).toMatchSnapshot()
})