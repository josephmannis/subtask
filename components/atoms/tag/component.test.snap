// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 22,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#3BA9ED",
          "borderRadius": 22,
          "display": "flex",
          "elevation": 12,
          "flexWrap": "nowrap",
          "minWidth": 100,
          "paddingBottom": 12,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 12,
          "shadowColor": "black",
          "shadowOpacity": 0.15,
          "shadowRadius": 12,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Test tags!
    </Text>
  </View>
</View>
`;
